### ISSUE TITLE
Inaccurate TypeScript example

### ISSUE THREAD

**Issue by PaulMorris** [automerge/old-automerge-website#16]
On the [Modeling Data](https://automerge.org/docs/cookbook/modeling-data/) page, under "TypeScript support", there is this example:
```
type D = { 
  count: Automerge.Counter,
  text: Automerge.Text,
  cards: [] 
}
let doc = Automerge.change<D>(Automerge.init(), (doc: D) => {
  doc.count = new Automerge.Counter()
  doc.text = new Automerge.Text()
  doc.cards = []
})
```
But then the top-level `doc` (the `let doc` one) is inaccurately typed as plain `D` because the return value of `Automerge.change` is actually an `Automerge.FreezeObject<D>`.

To simply correct the example:
```
let doc = Automerge.change<Automerge.FreezeObject<D>>(Automerge.init(), (doc: D) => {
```

Or (probably better for user ergonomics) the type declaration for `Automerge.change` could be changed so that the example can remain as it is.  Currently the type is like this:
```
function change<D, T = Proxy<D>>(doc: D, callback: ChangeFn<T>): D
```
With a return of plain `D`.  ~~Instead I think it could be this, changing `D` to `Doc<D>` in two places:~~ (Edit: this doesn't work fully, see comment below.)
```
function change<D, T = Proxy<D>>(doc: Doc<D>, callback: ChangeFn<T>): Doc<D>
```
(`Doc<D>` is shorthand for `Automerge.FreezeObject<D>`.)

That is what the signature of `Automerge.applyChanges` does (so there may be some inconsistency here in the type signatures):
```
function applyChanges<T>(doc: Doc<T>, changes: BinaryChange[]): [Doc<T>, Patch]
```

(This is with automerge version 1.0.1-preview.7 )

**Comment by PaulMorris** [comment]
After looking into this a bit more, here's a simpler way to do it.  I'll see if I can open a PR over on the automerge repo. (github is not letting me fork at the moment.)
```
function change<T>(doc: Doc<T>, callback: ChangeFn<T>): Doc<T>
```

**Comment by ept** [comment]
Closing this as it is fixed by automerge/automerge#479.
